src/client/api/init.c

```diff
@@ -195,19 +199,29 @@ daos_init(void)
 	if (rc != 0)
 		D_GOTO(out_agent, rc);
 
+	D_ALLOC_PTR(info);
+	if (info == NULL)
+		D_GOTO(out_job, rc);
+	rc = dc_get_attach_info(NULL, true, info, &resp);
+	if (rc != 0) {
+		D_FREE(info);
+		D_GOTO(out_job, rc);
+	}
+	dc_set_global_attach_info(info, resp);
+
 	/**
 	 * get CaRT configuration (see mgmtModule.handleGetAttachInfo for the
 	 * handling of NULL system names)
 	 */
 	rc = dc_mgmt_net_cfg(NULL);
```

Comment:
[Question] @mchaarawi, @frostedcmos, does anybody know why we have the `dc_mgmt_net_cfg_check` call below? How does `daos_eq_lib_init` "change net cfg"?

Reply:
From my reading, this has to do with 'FI_OFI_RXM_USE_SRX' (using shared receive queues), as this setting must match between clients and servers. 

daos_eq_lib_init() calls crt_init_opt() which then will do:

 550         if (prov == CRT_PROV_OFI_VERBS_RXM ||
 551             prov == CRT_PROV_OFI_TCP_RXM) {
 552                 /* Use shared receive queues to avoid large mem consumption */
 553                 d_setenv("FI_OFI_RXM_USE_SRX", "1", 0);

this can cause mismatched use_srx setting between clients and servers which dc_mgmt_net_cfg_check() checks.

Reply:
Thanks, @frostedcmos. This is quite eye-opening. :)